### DISCUSSION TITLE
Question on filtering results by distances between vectors

### DISCUSSION THREAD

**Discussion by clippered** [milvus-io/milvus#4445] [Q&A and General discussion] [answered]
HI,

In the example below, it returns the top 10 results with closest distance:
`query_hybrid = {
    "bool": {
        "must": [
            {
                "vector": {
                    "embedding": {"topk": 10, "query": [query_embedding], "metric_type": "IP"}
                }
            }
        ]
    }
}`

Is there an example of a query to return all results with distance greater than or less than a certain threshold? Maybe this example would like to see if the distance is greater than or equal 0.9:
`query_hybrid = {
    "bool": {
        "must": [
            {
                "vector": {
                    "embedding": {"distance_gte": 0.9, "query": [query_embedding], "metric_type": "IP"}
                }
            }
        ]
    }
}`

Thanks.

**Comment by sahuang** [comment, marked as answer]
Currently, there is no mechanism in Milvus to filter a range for distance. All you have to do is to add extra logic on top of DSL query that filters result by distances. This may require you to use a slightly bigger `topK`.

**Reply by clippered** [reply]
Thanks @sahuang . Actually that is my plan for now, to filter it after getting the results. But I think it would be a good feature so that it does not return a hundred results if none of them are above or below the range.

**Reply by sahuang** [reply]
This was a [plan](https://github.com/milvus-io/milvus/issues/1133), but there are a lot of details to consider. If the user sets too low a `nprobe`, for example, they might get nothing as the result. We might support range search in the future, depending on how well our current work goes. Thanks anyway for this suggestion!

**Reply by keotic** [reply]
This would be an excellent feature, FAISS is already supporting range search out of the box, didn't FB solved the nprobe problem?

**Reply by sahuang** [reply]
Faiss is more of a tool rather than a system, so they need not consider how users behave. When designing Milvus we need to also think of user experience, which means a raw implementation is not enough. I think nprobe is not even considered as a problem for Faiss :)

**Comment by clippered** [comment]
Nice. Thanks for that. I'll mark this answered then.
